From 6cb1d8f8cc5dbec4be7f4527db2f1bf632768f0e Mon Sep 17 00:00:00 2001
From: Jan Petrous <jan.petrous@nxp.com>
Date: Fri, 19 Feb 2021 13:52:57 +0100
Subject: [PATCH 75/84] dts: s32g27a: pfe: refactor interface HIF number

The ethernet property fsl,pfeng-hif-channel was moved
to the basic include file fsl-s32g274a.dtsi to not
duplicate them in all targets

Issue: ALB-6478
Signed-off-by: Jan Petrous <jan.petrous@nxp.com>
---
 arch/arm64/boot/dts/freescale/fsl-s32g274a-evb.dts  | 3 ---
 arch/arm64/boot/dts/freescale/fsl-s32g274a-rdb.dtsi | 2 --
 arch/arm64/boot/dts/freescale/fsl-s32g274a.dtsi     | 9 ++++++---
 3 files changed, 6 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-s32g274a-evb.dts b/arch/arm64/boot/dts/freescale/fsl-s32g274a-evb.dts
index 07289af22d03..6914acbeb05d 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32g274a-evb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-s32g274a-evb.dts
@@ -78,7 +78,6 @@
 };
 
 &pfe0_if {
-	fsl,pfeng-hif-channel = <0>;
 	phy-mode = "sgmii";
 	phy-handle = <&mdio_a_phy4>;
 };
@@ -86,7 +85,6 @@
 &pfe1_if {
 	/* RGMII port occupied by GMAC0 */
 	status = "disabled";
-	fsl,pfeng-hif-channel = <1>;
 	phy-mode = "rgmii";
 	phy-handle = <&mdio_c_phy4>;
 	pinctrl-names = "default";
@@ -94,7 +92,6 @@
 };
 
 &pfe2_if {
-	fsl,pfeng-hif-channel = <2>;
 	phy-mode = "rgmii";
 	fixed-link {
 		speed = <1000>;
diff --git a/arch/arm64/boot/dts/freescale/fsl-s32g274a-rdb.dtsi b/arch/arm64/boot/dts/freescale/fsl-s32g274a-rdb.dtsi
index 101bc7da8e4d..f9bce4923f81 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32g274a-rdb.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-s32g274a-rdb.dtsi
@@ -26,7 +26,6 @@
 };
 
 &pfe0_if {
-	fsl,pfeng-hif-channel = <0>;
 	phy-mode = "sgmii";
 	fixed-link {
 		speed = <2500>;
@@ -35,7 +34,6 @@
 };
 
 &pfe2_if {
-	fsl,pfeng-hif-channel = <2>;
 	phy-mode = "rgmii";
 	phy-handle = <&mdio_b_phy4>;
 	pinctrl-names = "default";
diff --git a/arch/arm64/boot/dts/freescale/fsl-s32g274a.dtsi b/arch/arm64/boot/dts/freescale/fsl-s32g274a.dtsi
index b55f9d3ad6ea..a5ed62c55a1a 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32g274a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-s32g274a.dtsi
@@ -102,7 +102,8 @@
 				      "tx_rmii", "tx_mii";
 			local-mac-address = [ 00 04 9F BE EF 00 ];
 			fsl,pfeng-if-name = "pfe0";
-			fsl,pfeng-emac-id = <0>; /* Phy IF (EMAC) id */
+			fsl,pfeng-emac-id = <0>; /* PFE_EMAC_0 */
+			fsl,pfeng-hif-channel = <0>; /* PFE_HIF_0 */
 			pfe0_mdio: mdio@0 {
 				compatible = "fsl,pfeng-mdio";
 				#address-cells = <1>;
@@ -125,7 +126,8 @@
 				      "tx_rmii", "tx_mii";
 			local-mac-address = [ 00 04 9F BE EF 01 ];
 			fsl,pfeng-if-name = "pfe1";
-			fsl,pfeng-emac-id = <1>; /* Phy IF (EMAC) id */
+			fsl,pfeng-emac-id = <1>; /* PFE_EMAC_1 */
+			fsl,pfeng-hif-channel = <1>; /* PFE_HIF_1 */
 			pfe1_mdio: mdio@0 {
 				compatible = "fsl,pfeng-mdio";
 				#address-cells = <1>;
@@ -148,7 +150,8 @@
 				      "tx_rmii", "tx_mii";
 			local-mac-address = [ 00 04 9F BE EF 02 ];
 			fsl,pfeng-if-name = "pfe2";
-			fsl,pfeng-emac-id = <2>; /* Phy IF (EMAC) id */
+			fsl,pfeng-emac-id = <2>; /* PFE_EMAC_2 */
+			fsl,pfeng-hif-channel = <2>; /* PFE_HIF_2 */
 			pfe2_mdio: mdio@0 {
 				compatible = "fsl,pfeng-mdio";
 				#address-cells = <1>;
-- 
2.17.1

